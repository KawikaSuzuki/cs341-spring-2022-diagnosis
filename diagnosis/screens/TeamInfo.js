import React from 'react';
import { Text, View, StyleSheet, Image, ScrollView, Dimensions } from 'react-native';

const TeamInfo = () => (
        <ScrollView style={{flex: 1}} contentContainerStyle={styles.mainContainer}>
            <Text style={styles.title}>Meet the Team</Text>
            <View>
                <Image
                    style={{ width: 200, height: 200}} 
                    source={require('./../assets/Andre-314x314.png')}
                />
                <Text style={styles.person}>Andr√© M. Mansoor, M.D.</Text>
            </View>
            <View>
                <Image
                   style={{ width: 200, height: 200}} 
                    source={require('./../assets/Sullivan.jpeg')}
                />
                <Text style={styles.person}>Peter Sullivan, M.D.</Text>
            </View>
            <View>
                <Image
                    style={{ width: 200, height: 200}} 
                    source={require('./../assets/jgn.jpg')}
                />
                <Text style={styles.person}>Joseph G. Nugent, M.H.S.</Text>
            </View>
            <View>
                <Image
                    style={{ width: 200, height: 200}} 
                    source={require('./../assets/ishita-sharma.jpeg')}
                />
                <Text style={styles.person}>Ishita Sharma, M.B.B.S.</Text>
            </View>
        </ScrollView>
);

const styles = StyleSheet.create({
    mainContainer: {
        flexGrow: 1,
        alignItems: "center",
        padding: 25,
        backgroundColor: "#BFAA8C",
    },
    title: {
        fontSize: 30,
        fontWeight: "bold",
        color: "red",
    },
    person: {
        fontWeight: "bold"
    }
});

export default TeamInfo;