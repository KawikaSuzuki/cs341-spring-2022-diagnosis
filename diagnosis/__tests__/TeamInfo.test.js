import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import { Linking } from 'react-native';
import renderer from 'react-test-renderer';
import TeamInfo from '../screens/TeamInfo';

jest.mock("react-native/Libraries/Linking/Linking", () => ({
    openURL: jest.fn(() => Promise.resolve("mockResolve")),
}));

describe('<TeamInfo />', function() {
    test('renders correctly', function() {
        const tree = renderer.create(<TeamInfo />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('social icon works', function() {
        const {getByTestId} = render(<TeamInfo />);
        const twitter = getByTestId('twitter');
        expect(twitter).toBeTruthy();
        fireEvent.press(twitter);
        expect(Linking.openURL).toHaveBeenCalledTimes(1);
    })
});