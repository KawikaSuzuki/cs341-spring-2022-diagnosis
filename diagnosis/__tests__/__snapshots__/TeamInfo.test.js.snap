// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeamInfo /> renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "backgroundColor": "#BFAA8C",
      "flexGrow": 1,
      "padding": 25,
      "paddingTop": 20,
    }
  }
  showsVerticalScrollIndicator={false}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "red",
          "fontSize": 30,
          "fontWeight": "bold",
        }
      }
    >
      Meet the Team
    </Text>
    <View>
      <Image
        source={1}
        style={
          Object {
            "alignSelf": "center",
            "height": 200,
            "width": 200,
          }
        }
      />
      <Text
        style={
          Object {
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Andr√© M. Mansoor, M.D.
        

      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4267B2",
                "borderRadius": 30,
                "flexDirection": "row",
                "height": 52,
                "justifyContent": "center",
                "margin": 7,
                "shadowColor": "rgba(0,0,0, .4)",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 1,
                "shadowRadius": 1,
                "width": 52,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#517fa4",
                "borderRadius": 30,
                "flexDirection": "row",
                "height": 52,
                "justifyContent": "center",
                "margin": 7,
                "shadowColor": "rgba(0,0,0, .4)",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 1,
                "shadowRadius": 1,
                "width": 52,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#bb0000",
                "borderRadius": 30,
                "flexDirection": "row",
                "height": 52,
                "justifyContent": "center",
                "margin": 7,
                "shadowColor": "rgba(0,0,0, .4)",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 1,
                "shadowRadius": 1,
                "width": 52,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00aced",
                "borderRadius": 30,
                "flexDirection": "row",
                "height": 52,
                "justifyContent": "center",
                "margin": 7,
                "shadowColor": "rgba(0,0,0, .4)",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 1,
                "shadowRadius": 1,
                "width": 52,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View>
      <Image
        source={1}
        style={
          Object {
            "alignSelf": "center",
            "height": 200,
            "width": 200,
          }
        }
      />
      <Text
        style={
          Object {
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Peter Sullivan, M.D.
        

      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#00aced",
            "borderRadius": 30,
            "flexDirection": "row",
            "height": 52,
            "justifyContent": "center",
            "margin": 7,
            "shadowColor": "rgba(0,0,0, .4)",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
            "width": 52,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View>
      <Image
        source={1}
        style={
          Object {
            "alignSelf": "center",
            "height": 200,
            "width": 200,
          }
        }
      />
      <Text
        style={
          Object {
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Joseph G. Nugent, M.H.S.
        

      </Text>
    </View>
    <View>
      <Image
        source={1}
        style={
          Object {
            "alignSelf": "center",
            "height": 200,
            "width": 200,
          }
        }
      />
      <Text
        style={
          Object {
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Ishita Sharma, M.B.B.S.
        

      </Text>
    </View>
  </View>
</RCTScrollView>
`;
